test/integration/deploy_trigger_test.go

```diff
@@ -197,8 +206,19 @@ func TestTriggers_imageChange(t *testing.T) {
 		t.Fatalf("Expected deployment annotated with deploymentConfig '%s', got '%s'", e, a)
 	}
 
+	// expecting RC update from new to pending
```

Comment:
Is it safe to rely on a specific order of events without any others interleaving?  It feels like interleaving could happen.